lib/orderbook/OrderBook.ts

```diff
@@ -96,12 +102,22 @@ class OrderBook extends EventEmitter {
     return this.addOwnOrder(order);
   }
 
-  public addMarketOrder = (order: orders.MarketOrder) => {
+  public addMarketOrder = (order: orders.OwnMarketOrder): matchingEngine.MatchingResult => {
     const price = order.quantity > 0 ? Number.MAX_VALUE : 0;
     return this.addOwnOrder({ ...order, price }, true);
```

Comment:
I think we should verify `localId` uniqueness in the beginning of `addOwnOrder` method. Client shouldn't place a new order with an id which already exists. At the moment the later `localId` will override the previous one (`addOrder`) for the remaining order. swaps updates would also be problematic. And at the moment it creates a bug that I can't cancel an order if I re-used it's `localId`.